### ISSUE TITLE
Will you be supporting the ToDataSourceResult extensions methods in MVC for angular2 grids ?

### ISSUE THREAD

**Issue by paulstringerdb** [telerik/kendo-angular#158]
Previously we have used older Kendo controls and found the "ToDataSourceResult" extension method to be very beneficial with our code. We don't use ODATA endpoints and cant configure grids in that way, so we make use of IQueryable a lot.

Will you be planning to offer this kind of support for those of us "stuck" in the MVC world and who will require an "out of the box" solution for grid Ordering/Sorting/Filtering as we previously had. Writing our own expressions for all this could be so time consuming.

**Comment by rkonstantinov** [comment]
Indeed, we are considering adding a way to serialize the state descriptors to the format required by the UI for ASP.NET MVC ToDataSourceResult extension methods in near future.

**Comment by paulstringerdb** [comment]
Fantastic ! This will be a huge win for us...

**Comment by phani257** [comment]
@rkonstantinov  I am trying to build a new app using ASP.NET MVC but wanted to use kendo-angular. All my angular dependencies are set properly, but when I referred just the kendo-angular-button and run the app, it throws cannot load http://localhost:59086/@progress/kendo-angular-buttons.

**Comment by rkonstantinov** [comment]
@phani257 I'm afraid that I'm not sure what is causing the issue on your side nor how your question is related to the topic of this discussion. Therefore, if you believe that the issue is caused by our component(s), please open a separate issue where to provide a runnable sample demonstrating the behavior.

**Comment by phani257** [comment]
@rkonstantinov  This is not related to your component. But just want to check with you whether you have heard any similar case which can help to resolve my issue. I am not sure how many are implementing the way that we are using now. kendo-angular2 components on ASP.NET MVC framework.

**Comment by sgvalsamis** [comment]
@rkonstantinov 
I saw in changelog v0.5.0 (03/22/17) of the Data Query Component, that you have added support for UI for ASP.NET MVC DataSourceRequest and ToDataSourceResult.
Can you provide some instruction to implement this senario, or a sample(much better) ?
Thanks

**Comment by rkonstantinov** [comment]
In order to serialize the operation descriptors you should use the [`toDataSourceRequestString`](http://www.telerik.com/kendo-angular-ui/components/dataquery/api/toDataSourceRequestString/), then if grouping is used the [`translateDataSourceResultGroups`](http://www.telerik.com/kendo-angular-ui/components/dataquery/api/translateDataSourceResultGroups/) should be used to convert the grouped server response.
```js
import {
     toDataSourceRequestString,
     translateDataSourceResultGroups,
     translateAggregateResults
} from '@progress/kendo-data-query';

export class Service {
 private BASE_URL: string = '...';

 constructor(private http: Http) { }

 // omitted for brevity..

 private fetch(state: DataSourceRequestState): Observable<DataResult> {
  const queryStr = `${toDataSourceRequestString(state)}`; //serialize the state
  const hasGroups = state.group && state.group.length;

  return this.http
      .get(`${this.BASE_URL}?${queryStr}`) //send the state to the server
      .map(response => response.json())
      .map(({Data, Total, AggregateResults}) => // process the response
          (<GridDataResult>{
              //if there are groups convert them to compatible format
              data: hasGroups ? translateDataSourceResultGroups(Data) : Data,
              total: Total,
              // convert the aggregates if such exists
              aggregateResult: translateAggregateResults(AggregateResults)
          })
      );
 }
}
```

**Comment by sgvalsamis** [comment]
@rkonstantinov 
thanks for your response.

I can not find where "DataSourceRequestState" is exported from.
It is not exported by @progress/kendo-data-query.
I have install kendo-data-query using command line npm --save @progress/kendo-data-query
Did I missed something?

thanks!

**Comment by rkonstantinov** [comment]
Indeed, it seems that it is not exported from the package. We will  try to address it soon as possible.

**Comment by sgvalsamis** [comment]
Thanks,
looking forward to your next build

**Comment by rkonstantinov** [comment]
@sgvalsamis The type should be available in the 0.5.1 version of `kendo-data-query` package

**Comment by sgvalsamis** [comment]
Thank you!!!!
